src/lib/bootloader/grub2pwd.rb

```diff
@@ -1,63 +1,146 @@
 require "yast"
 
-# class is responsible for detection, encryption and writing of grub2 password protection
-class GRUB2Pwd
-  YAST_BASH_PATH = Yast::Path.new(".target.bash_output")
-  PWD_ENCRYPTION_FILE = "/etc/grub.d/42_password"
+Yast.import "Stage"
 
-  def used?
-    Yast.import "FileUtils"
+module Bootloader
+  # class is responsible for detection, encryption and writing of grub2 password protection
+  class GRUB2Pwd
+    # @!attribute used
+    #   @return [bool] specifies if password protection enabled
+    #
+    # @!attribute unrestricted
+    #   @return [bool] specifies if unrestricted password protection should be used
```

Comment:
Yes, but Boolean is a YARD convention whereas bool is not: http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md#Declaring_Types

Yes, "unrestricted" is an official grub2 term, but it is local to a single entry whereas we use it in a global meaning. Even if I read the docs I will not know what it means is SUSE: https://www.gnu.org/software/grub/manual/html_node/Security.html#Security